applications/mne_analyze/extensions/dataviewer/dataviewer.cpp

```diff
@@ -30,105 +30,179 @@
 //=============================================================================================================
-// DEFINE GLOBAL METHODS
+// DEFINE MEMBER METHODS
 //=============================================================================================================
 
+DataViewer::DataViewer()
+    : m_pControl(Q_NULLPTR)
+    , m_pDataViewerControl(Q_NULLPTR)
+{
+
+}
+
+
+//*************************************************************************************************************
+
+DataViewer::~DataViewer()
+{
+
+}
+
+
+//*************************************************************************************************************
+
+QSharedPointer<IExtension> DataViewer::clone() const
+{
+    QSharedPointer<DataViewer> pDataViewerClone = QSharedPointer<DataViewer>::create();
+    return pDataViewerClone;
+}
+
 
 //*************************************************************************************************************
-//=============================================================================================================
-// DEFINE MEMBER METHODS
-//=============================================================================================================
 
-SurfaceSettings::SurfaceSettings(FSLIB::Surface *pSurface)
-: m_pSurface(pSurface)
+void DataViewer::init()
 {
+    m_pDataViewerControl = new DataViewerControl;
+
+    updateListWidget();
+
+    connect(m_analyzeData.data(), &AnalyzeData::newModelAvailable,
+            this, &DataViewer::updateListWidget);
+    connect(m_analyzeData.data(), &AnalyzeData::modelPathChanged,
+            this, &DataViewer::updateListWidget);
+    connect(m_analyzeData.data(), &AnalyzeData::modelRemoved,
+            this, &DataViewer::updateListWidget);
+
 }
 
 
 //*************************************************************************************************************
 
-qint32 SurfaceSettings::getHemi() const
+void DataViewer::unload()
 {
-    return m_pSurface->hemi();
 }
 
 
 //*************************************************************************************************************
 
-QString SurfaceSettings::getSurfaceType() const
+QString DataViewer::getName() const
 {
-    return m_pSurface->surf();
+    return "DataViewer";
 }
 
 
 //*************************************************************************************************************
 
-Eigen::Vector3f SurfaceSettings::getOffset() const
+QMenu *DataViewer::getMenu()
 {
-    return m_pSurface->offset();
+    return Q_NULLPTR;
 }
 
 
 //*************************************************************************************************************
 
-QString SurfaceSettings::getFilePath() const
+QDockWidget *DataViewer::getControl()
 {
-    return m_pSurface->filePath();
+    if(!m_pControl) {
+        m_pControl = new QDockWidget(tr("Data View Control"));
+        m_pControl->setAllowedAreas(Qt::LeftDockWidgetArea | Qt::RightDockWidgetArea);
+        m_pControl->setMinimumWidth(180);
```

Comment:
This is not necessary, since it was agreed that size constraints of GUI widgets should only be defined in .ui files.